sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/FunctionToolCallingSample.java

```diff
@@ -0,0 +1,248 @@
+                new ChatCompletionsOptions(chatRequestMessages));
+        getContentFromMessageResponse(chatCompletionsAnswer);
+    }
+
+    private static String getContentFromMessageResponse(ChatCompletions chatCompletions) {
+        List<ChatChoice> choices = chatCompletions.getChoices();
+        ChatResponseMessage message = choices.get(0).getMessage();
+        System.out.printf("Message: %s.%n", message.getContent());
+        return message.getContent();
+    }
+
+    private static FunctionParameters getCurrentWeatherFunctionParameters() {
+        FunctionProperties location = new FunctionProperties()
+                .setType("string")
+                .setDescription("The city and state, e.g. San Francisco, CA");
+
+        FunctionProperties format = new FunctionProperties()
+                .setType("string")
+                .setEnumString(Arrays.asList("celsius", "fahrenheit"))
+                .setDescription("The temperature unit to use. Infer this from the users location.");
+
+        Map<String, FunctionProperties> props = new HashMap<>();
+        props.put("location", location);
+        props.put("format", format);
+
+        return new FunctionParameters()
+                .setType("object")
+                .setRequiredPropertyNames(Arrays.asList("location", "format"))
+                .setProperties(props);
+    }
+
+    private static FunctionParameters getNDayWeatherForecastFunctionParameters() {
+        FunctionProperties location = new FunctionProperties()
+                .setType("string")
+                .setDescription("The city and state, e.g. San Francisco, CA");
+
+        FunctionProperties format = new FunctionProperties()
+                .setType("string")
+                .setEnumString(Arrays.asList("celsius", "fahrenheit"))
+                .setDescription("The temperature unit to use. Infer this from the users location.");
+
+        FunctionProperties numDays = new FunctionProperties()
+                .setType("integer")
+                .setDescription("The number of days to forecast");
+
+        Map<String, FunctionProperties> props = new HashMap<>();
+        props.put("location", location);
+        props.put("format", format);
+        props.put("num_days", numDays);
+
+        return new FunctionParameters()
+                .setType("object")
+                .setRequiredPropertyNames(Arrays.asList("location", "format", "num_days"))
+                .setProperties(props);
+    }
+
+    private static List<ChatRequestMessage> handleFunctionToolResponse(List<ChatChoice> choices,
+                                                                      List<ChatRequestMessage> chatMessages) {
+        for (ChatChoice choice : choices) {
+            ChatResponseMessage choiceMessage = choice.getMessage();
+
+            List<ChatCompletionsToolCall> toolCalls = choiceMessage.getToolCalls();
+            if (toolCalls != null) {
```

Comment:
I would rather add a check here for `finish_reason.TOOL_CALL` as well (or instead)

Reply:
I've written sync/async samples for this in case you need to check that out. [Link](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/usage/GetChatCompletionsToolCallSample.java)

Reply:
Also streaming (it's not pretty)

Reply:
I saw there is a possible that having "finish_reason is STOPPED" but still have toolCalls values. 

![image](https://github.com/Azure/azure-sdk-for-java/assets/45607042/3e87a3f7-0838-4283-a26a-8be20ae4793b)

Reply:
Are those the choices from the `choiceMessage`. I am not sure I can tell exactly from the screen shot 🤔 I wasn't aware we could have a `finish_reason` other than `tool_call` in this case.